lib/apartment/adapters/abstract_adapter.rb

```diff
@@ -140,6 +148,7 @@ def connection_switch!(config, without_keys: [])
 
         unless Apartment.connection_handler.retrieve_connection_pool(config[:name])
           Apartment.connection_handler.establish_connection(config)
+          Apartment.establish_connection(config)
```

Comment:
Either this architecture is unfortunate, or the naming here is off, but this feels redundant.

Reply:
So this hack was unfortunately the solution I came to when I realized that Apartment was not always switching databases when I expected it to. It's possible that some of the other changes I've since made to `connection_specification_name` would allow us to only call one `establish_connection` here.

Reply:
is it worth trying to resolve?